src/pal/src/synchobj/mutex.cpp

```diff
@@ -347,25 +355,20 @@ CorUnix::InternalCreateMutex(
         }
     }
 
-    //
-    // pobjMutex is invalidated by the call to RegisterObject, so NULL it
-    // out here to ensure that we don't try to release a reference on
-    // it down the line.
-    //
-    
-    pobjMutex = NULL;
     *phMutex = hMutex;
+    hMutex = nullptr;
+    pobjMutex = nullptr;
 
 InternalCreateMutexExit:
 
-    if (NULL != pobjMutex)
+    _ASSERTE(pobjRegisteredMutex == nullptr);
+    if (hMutex != nullptr)
     {
-        pobjMutex->ReleaseReference(pthr);
+        g_pObjectManager->RevokeHandle(pthr, hMutex);
     }
-
-    if (NULL != pobjRegisteredMutex)
+    else if (NULL != pobjMutex)
```

Comment:
Also, can you change the order to `pobjMutex != nullptr` so it's consistent with the previous if?